src/main/java/nus/climods/logic/parser/ParserUtil.java

```diff
@@ -88,4 +89,21 @@ public static Optional<String> parseFacultyCode(String input) {
         return Optional.of(input);
     }
 
+    /**
+     * Parse semester details according to user input
+     * @param input Input string from user containing semester details
+     * @return Optional of String representing semester number. Returns empty Optional if user input is not a valid
+     *      semester.
+     */
+    public static Optional<String> parseSemesterType(String input) {
```

Comment:
nit: Should we just return an Optional of Integer for the semester number instead? In the end the String is being converted into an integer when checking inside UserModule. 

Was thinking it might even be good to make Semester numbers an Enum as well at least when storing inside UserModule, which makes checking inside SavedModuleCard less error prone. What do you all think?

Reply:
hmm yes I agree that enum is better too...

Reply:
Can check out ModuleCondensed.java, there's a SemesterEnum class there which I think should be abstracted out and used in all other classes